### ISSUE TITLE
Ignore pragma when considering E501

### ISSUE THREAD

**Issue by baggiponte** [astral-sh/ruff#3894]
I wrote this snippet of code and the `<- !` line triggers `E501`.

```python
def load_dynamodb(fileobj: IO | str | bytes | bytearray) -> dict[str, Any]:
    """Loads a dynamodb json fileobject."""
    if isinstance(fileobj, (str, bytes, bytearray)):
        return json.loads(fileobj, object_hook=_dynamo_hook)  # type: ignore [no-any-return] <- !
    return json.load(fileobj, object_hook=_dynamo_hook)  # type: ignore [no-any-return]
```

Since what triggers the warning is the pragma, I wanted to ask whether you believe that pragmas should not trigger `E501` - something that Google has recently done with their [`pyink`](https://github.com/google/pyink#what-are-the-main-differences) Python code formatter.

**Comment by charliermarsh** [comment]
Yeah that's an interesting question. I tend to agree, but let me leave this open for a bit to see if any other opinions come in.

**Comment by bryanforbes** [comment]
`flake8-bugbear` recently changed `B950` to ignore `noqa` and `type: ignore` comments when determining whether to report the error or not, so there's some precedence for ignoring pragmas

**Comment by charliermarsh** [comment]
Oh that's great to know. Honestly that might be enough of a justification to just ship it :)

**Comment by charliermarsh** [comment]
We made this change recently. It'll be out in the next release.
